qemu/tests/ovs_host_vlan.py

```diff
@@ -0,0 +1,249 @@
+        if status != 0:
+            err = "Fail to start netserver in VM."
+            err += " Command output %s" % output
+            raise error.TestError(err)
+
+        txt = "Run netperf client in VM %s" % vms[1]
+        error.context(txt, logging.info)
+        test_duration = int(params.get("netperf_test_duration", 60))
+        test_protocol = params.get("test_protocol")
+        netperf_cmd = params.get("netperf_cmd")
+        netperf_cmd = os.path.join(netperf_path, netperf_cmd)
+        cmd = netperf_cmd % (test_duration, ips[0])
+        if test_protocol:
+            cmd += " -t %s" % test_protocol
+        cmd_timeout = test_duration + 20
+        status, output = sessions[1].cmd_status_output(cmd, timeout=cmd_timeout)
+        if status != 0:
+            err = "Fail to run netperf test between %s and %s." % (vms[0],
+                                                                   vms[1])
+            err += " Command output:\n%s" % output
+            raise error.TestFail(err)
+
+        if params.get("file_transfer_test", "yes") == "yes":
+            filesize = int(params.get("filesize", 1024))
+            file_create_timeout = int(params.get("file_create_timeout", 720))
+            file_path = params.get("file_path", "/var/tmp/src_file")
+
+            txt = "Create %s MB file %s in %s" % (filesize, file_path, vms[0])
+            error.context(txt, logging.info)
+            create_file_in_guest(session=sessions[0], file_path=file_path,
+                                 size=filesize, os_type=os_type,
+                                 timeout=file_create_timeout)
+
+            txt = "Transfer file %s between guests in same vlan. " % file_path
+            txt += "%s -> %s" % (vms[0], vms[1])
+            error.context(txt, logging.info)
+            password = params.get("password", "redhat")
+            username = params.get("username", "root")
+            f_timeout = int(params.get("file_transfer_timeout", 1200))
+            shell_port = params.get("shell_port", "22")
+            data_port = params.get("nc_file_transfer_port", "9000")
+            log_file = "file_transfer_from_%s_to_%s.log" % (ips[0], ips[1])
+            if os_type == "linux":
+                remote.nc_copy_between_remotes(ips[0], ips[1], shell_port,
+                                               password, password,
+                                               username, username,
+                                               file_path, file_path,
+                                               d_port=data_port,
+                                               timeout=2,
+                                               check_sum=True,
+                                               s_session=sessions[0],
+                                               d_session=sessions[1],
+                                               file_transfer_timeout=f_timeout)
+    finally:
+        ovs_del_cmd = "ovs-vsctl del-br %s" % bridge_name
```

Comment:
On 01/13/2015 11:19 AM, Xu Tian wrote:
>
> In qemu/tests/ovs_host_vlan.py
> <https://github.com/autotest/tp-qemu/pull/241#discussion-diff-22841374>:
>
> > +            shell_port = params.get("shell_port", "22")
> > +            data_port = params.get("nc_file_transfer_port", "9000")
> > +            log_file = "file_transfer_from_%s_to_%s.log" % (ips[0], ips[1])
> > +            if os_type == "linux":
> > +                remote.nc_copy_between_remotes(ips[0], ips[1], shell_port,
> > +                                               password, password,
> > +                                               username, username,
> > +                                               file_path, file_path,
> > +                                               d_port=data_port,
> > +                                               timeout=2,
> > +                                               check_sum=True,
> > +                                               s_session=sessions[0],
> > +                                               d_session=sessions[1],
> > +                                               file_transfer_timeout=f_timeout)
> > +    finally:
> > +        ovs_del_cmd = "ovs-vsctl del-br %s" % bridge_name
>
> Do we need to stop vm first before drop ovs bridge.
>
Until now did not find it break anything.  VM should be killed after
this command soon.


> —
> Reply to this email directly or view it on GitHub
> <https://github.com/autotest/tp-qemu/pull/241/files#r22841374>.
>